coalib/settings/ConfigurationGathering.py

```diff
@@ -22,24 +23,68 @@
                           '* add `-I` to suppress any use of config files\n')
 
 
-def aspectize_sections(sections):
+def aspectize_sections(sections, acquire_settings):
     """
     Search for aspects related setting in a section, initialize it, and then
     embed the aspects information as AspectList object into the section itself.
 
     :param sections:  List of section that potentially contain aspects setting.
     :return:          The new sections.
     """
-    for section_name, section in sections.items():
-        section.aspects = extract_aspects_from_section(section)
-        if section.aspects is not None and len(section.get('bears')):
-            logging.warning("'aspects' and 'bears' configuration is detected "
-                            "in section '{}'. Aspect-based configuration will "
-                            'takes priority and will overwrite any '
-                            'explicitly listed bears'.format(section_name))
+    for _, section in sections.items():
+        if verify_aspect_config(section, acquire_settings):
+            section.aspects = extract_aspects_from_section(section)
+        else:
+            section.aspects = None
     return sections
 
 
+def verify_aspect_config(section, acquire_settings):
+    """
+    Verify if a section contain required setting to run in aspects mode.
+
+    In case of missing language setting, it will require an input.
+
+    :param section:         The section that potentially contain aspect
+                            setting.
+    :param acquire_setting: Method used to acquire missing setting.
+    :return:                The validity of section.
+    """
+    aspects = section.get('aspects')
+    aspects_language = section.get('language')
+
+    if not len(aspects):
+        return False
+
+    if not len(aspects_language):
+        logging.warning('Setting `language` is not found in section `{}`. '
+                        'Usage of aspect-based configuration must include '
+                        'language information.'.format(section.name))
+        language_dict = acquire_settings(
+            LogPrinter(),
+            {'language': ['The file\'s language in section `{}`'
+                          .format(section.name),
+                          'all aspect bears']},
+            section)
+        aspects_language = language_dict['language']
+        section.append(Setting('language', aspects_language))
+
+    try:
+        language(aspects_language)
+    except ValueError as exc:
+        logging.warning('{} Coala will fallback to using `bears` setting.'
```

Comment:
What if there is no `bears` setting?

Reply:
It will be catched by `warn_nonexistent_targets` (if entire .coafile doesn't have bears/aspects). If only 1 section doesn't have it, coala will skip it.

Reply:
Well it will be a bit misleading if `bears is not present, but it still skip the whole section.

Reply:
are you sure?  A section has `aspects`, no `bears` and an invalid value for `language`.  Is this error message here going to be shown?  If so, the error message text is wrong.